test/Constraints/static_members_on_protocol_in_generic_context.swift

```diff
@@ -243,30 +249,47 @@ test(.generic(42).other)
 test(.generic(S())) // expected-error {{contextual member reference to static method 'generic' requires 'Self' constraint in the protocol extension}}
 test(.generic(G<Int>())) // expected-error {{contextual member reference to static method 'generic' requires 'Self' constraint in the protocol extension}}
 test(.genericWithReqs([S()])) // expected-error {{contextual member reference to static method 'genericWithReqs' requires 'Self' constraint in the protocol extension}}
+// expected-error@-1 {{static method 'genericWithReqs' requires that '[any Q]' conform to 'Q'}}
+test(.genericWithReqs([S()]).doesntExist) // expected-error {{contextual member reference to static method 'genericWithReqs' requires 'Self' constraint in the protocol extension}}
+// expected-error@-1 {{value of type 'any Q' has no member 'doesntExist'}}
+// expected-error@-2 {{static method 'genericWithReqs' requires that '[any Q]' conform to 'Q'}}
 test(.genericWithReqs([42]))
 // expected-error@-1 {{contextual member reference to static method 'genericWithReqs' requires 'Self' constraint in the protocol extension}}
+// expected-error@-2 {{static method 'genericWithReqs' requires that '[any Q]' conform to 'Q'}}
+// expected-error@-3 {{cannot convert value of type 'Int' to expected element type 'any Q'}}
 test(.genericWithReqs(()))
 // expected-error@-1 {{type '()' cannot conform to 'Collection'}}
-// expected-note@-2 {{only concrete types such as structs, enums and classes can conform to protocols}}
+// expected-error@-2 {{type '()' cannot conform to 'Q'}}
 // expected-error@-3 {{contextual member reference to static method 'genericWithReqs' requires 'Self' constraint in the protocol extension}}
+// expected-note@-4 2 {{only concrete types such as structs, enums and classes can conform to protocols}}
 
 test_combo(.doesntExist) // expected-error {{reference to member 'doesntExist' cannot be resolved without a contextual type}}
 test_combo(.doesnt.exist()) // expected-error {{reference to member 'doesnt' cannot be resolved without a contextual type}}
 test_combo(.invalidProp)
 // expected-error@-1 {{contextual member reference to static property 'invalidProp' requires 'Self' constraint in the protocol extension}}
+test_combo(.invalidProp.doesntExist) //FIXME Should be reporting that Int has no doesntExist if P&Q lookup fixed
```

Comment:
Minor nit, this allows it to be highlighted in Xcode:

```suggestion
test_combo(.invalidProp.doesntExist) // FIXME: Should be reporting that Int has no doesntExist if P&Q lookup fixed
```

Same goes for the others below